### PR TITLE
Clean up and standardize variables in tests

### PR DESCRIPTION
**Pull request by robin-checkmk**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

The integration tests use inconsistent variable naming, which is ranging from suboptimal, to confusing and beyond.

## What is the new behavior?

Variable names are aligned with collection standards and linting now ensures variables meet the requirements moving forward.

## Other information

There are no functional changes to the integration tests whatsoever. It is merely a significant renaming of variables.

### CODE REVIEW

**Comment by robin-checkmk** [comment]
@msekania I cannot expect this from you and I don't. But if you can spare a few minutes, I appreciate any and every pair of eyes, that double-checks, if I broke testing logic or if I missed something. But a "no" to this request is more than fine. :)

**Comment by msekania** [comment]
@robin-checkmk, yes sure, no problem.
I'll have a look late today if it's fine

**Review by msekania — COMMENTED** [review]
After just visual code reviewing.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -131,27 +134,28 @@ You can also run a subset by mentioning them as follows. See `tests/integration/
...
-
-    cd roles/server/
-    molecule test -s 2.2.0
-    molecule test -s 2.3.0
-
 ### Unit
 
-Currently we're only unit testing the inventory plugin.
-Like with integration tests, we recommend testing inside docker, so modification of your local system isn't necessary.
+[Ansible Unit Tests](https://docs.ansible.com/ansible/latest/dev_guide/testing_unit.html) run unit tests created by the maintainers of the respective modules, to ensure the code works as intended. The `ansible-test` tool typically comes along with your Ansible installation (e.g. if you use the `requirements.txt` of this project).
```
[the remark this answers is not in the document]
Comment (robin-checkmk) [reply CONTRIBUTING.md:139]: Dang it, there is an `s` missing: https://docs.ansible.com/ansible/latest/dev_guide/testing_units.html

I will fix it.

File: tests/integration/targets/downtime/tasks/test.yml [hunk 2]
```diff
@@ -305,9 +305,9 @@
     service_descriptions:
       - "Check_MK"
     state: absent
-  register: result
-  failed_when: result.changed == False
-  loop: "{{ checkmk_hosts }}"
+  register: __checkmk_var_result_delete
+  failed_when: __checkmk_var_result_delete.changed == False
```
[the remark this answers is not in the document]
Comment (robin-checkmk) [reply tests/integration/targets/downtime/tasks/test.yml:309]: I like it, done!

File: tests/integration/targets/user/tasks/test.yml [hunk 3]
```diff
@@ -245,12 +222,6 @@
...
+  register: __checkmk_var_result_users_redelete
   loop: "{{ checkmk_var_users_create }}"
-
-- name: "{{ outer_item.version }} - {{ outer_item.edition | upper }} - Fail if changed."  # noqa no-handler
-  ansible.builtin.fail:
-    msg: "Rule changed!"
-  when: "rule_result.changed"
-  delegate_to: localhost
-  run_once: true # noqa run-once[task]
+  failed_when: __checkmk_var_result_users_redelete-changed
```
[the remark this answers is not in the document]
Comment (robin-checkmk) [reply tests/integration/targets/user/tasks/test.yml:227]: At some point refactoring stuff, you just overlook stupid typos like this. Thanks for catching it!

File: tests/integration/targets/contact_group/tasks/test.yml [hunk 4]
```diff
@@ -15,30 +15,30 @@
     site: "{{ outer_item.site }}"
     automation_user: "{{ checkmk_var_automation_user }}"
     automation_secret: "{{ checkmk_var_automation_secret }}"
-    customer: "{{ (customer != None) | ternary(customer, omit) }}"  # See PR #427
+    customer: "{{ (checkmk_var_customer != None) | ternary(checkmk_var_customer, omit) }}"  # See PR #427
```
[the remark this answers is not in the document]
Comment (robin-checkmk) [reply tests/integration/targets/contact_group/tasks/test.yml:18]: We had a discussion [here](https://github.com/Checkmk/ansible-collection-checkmk.general/pull/427#discussion_r1335762098), and I still do not really understand, why the approach you suggest did not work, but I remember to have seen it myself. We can give it another try though, to see, if the issue is still there.

[the remark this answers is not in the document]
Comment (msekania) [reply tests/integration/targets/contact_group/tasks/test.yml:18]: OK, I see the issues.
Then it should stay as it is.

**Review by msekania — CHANGES_REQUESTED** [review]
@robin-checkmk I think that's it

File: tests/integration/targets/lookup_sites/tasks/test.yml [hunk 5]
```diff
@@ -16,9 +16,9 @@
 
 - name: "{{ outer_item.version }} - {{ outer_item.edition | upper }} - Verify local connection."
   ansible.builtin.debug:
-    msg: "{{ extensions }}"
+      msg: "{{ __checkmk_var_sites }}"
```
[the remark this answers is not in the document]
Comment (robin-checkmk) [reply tests/integration/targets/lookup_sites/tasks/test.yml:19]: Fun fact: `ansible-lint` does not care about this, as long as the indentation is consistent. I fixed it anyways, thanks!

File: tests/integration/targets/lookup_rules/tasks/test.yml [hunk 6]
```diff
@@ -93,25 +93,25 @@
                   }}"
   delegate_to: localhost
   run_once: true  # noqa run-once[task]
-  loop: "{{ cpu_load_ruleset.rules }}"
+  loop: "{{ __checkmk_var_cpu_load_ruleset.__checkmk_var_one_ruleset }}"
```
[the remark this answers is not in the document]
Comment (robin-checkmk) [reply tests/integration/targets/lookup_rules/tasks/test.yml:96]: It took me way to long to understand how to actually fix this, as after my initial refactoring it did not work anymore.
The problem is, that `rules` here was just the name of the registered variable in line `27`.
As that name is changed as well, we need to update it here accordingly.

Or did you see something, I did not see?

[the remark this answers is not in the document]
Comment (msekania) [reply tests/integration/targets/lookup_rules/tasks/test.yml:96]: @robin-checkmk 
`.__checkmk_var_one_ruleset`  ->  `.rules`

[the remark this answers is not in the document]
Comment (msekania) [reply tests/integration/targets/lookup_rules/tasks/test.yml:96]: @robin-checkmk
OK, I see now

**Review by lgetwan — APPROVED** [review]
Only did a rough scan-through, but looks good to me!

**Pull request merged**